File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/usecase/DeleteHelpcardUseCase.java
package space.lopatkin.spb.helpboardgamecard.domain.usecase;

import space.lopatkin.spb.helpboardgamecard.domain.repository.HelpcardRepository;
import space.lopatkin.spb.helpboardgamecard.model.Helpcard;

public class DeleteHelpcardUseCase {

    private HelpcardRepository repository;

    public DeleteHelpcardUseCase(HelpcardRepository repository) {
        this.repository = repository;
    }

    public void execute(Helpcard helpcard) {
        repository.delete(helpcard);
    }

}

File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/usecase/DeleteHelpcardUseCase.kt
package space.lopatkin.spb.helpboardgamecard.domain.usecase

import space.lopatkin.spb.helpboardgamecard.domain.model.Helpcard
import space.lopatkin.spb.helpboardgamecard.domain.repository.AppRepository

class DeleteHelpcardUseCase(private val repository: AppRepository) {

    fun execute(helpcard: Helpcard?) {
        repository.delete(helpcard = helpcard!!)
    }

}
